libraries/autoload.rb

```diff
@@ -1,3 +1,7 @@
+unless Gem::Requirement.new(">= 12.0").satisfied_by?(Gem::Version.new(Chef::VERSION))
+  raise "The compat_resources cookbook does not support chef versions older than Chef 12.0.0"
```

Comment:
@lamont-granquist consider simply logging an error and returning to the caller instead of raising?  Most folks who use chef 11.x clients would rather a no-op than a failing chef-client run.